internal/app/secretless/handlers/pg/protocol/ssl.go

```diff
@@ -0,0 +1,149 @@
+package protocol
```

Comment:
General comment that relates to this file and to backend.go... I'm writing this after having written all the other comments, and it's something larger that's still bothering me...

The "division of labor" among these files isn't clear... it's doesn't feel intuitive.  What are the responsibilities _really_?  This could use more design and thought imo, because it doesn't "stick" (and the names don't match with things really are).  I find myself hopping back to definitions a few minutes... even 30s after I've read them.

A few examples... what is currently called `ResolveTLSConfig` appears to be a mapping from user supplied "QueryStrings" to options in the native `tlsConf` config object.  That is, it's essentially a pure transformation function, and it's related to user input as it comes into secretless.  Yet it's currently house in the `protocol.ssl.go`, and I'd say it's a stretch to call this function part of the ssl protocol.  

OTOH, in `pg.backend.go`, we have snippets like this:

```go
		message := protocol.NewMessageBuffer([]byte{})
		message.WriteInt32(8)
		message.WriteInt32(protocol.SSLRequestCode)
```
which is a byte level details of the pg ssl protocol.

There are other similar bit of confusion.  I don't have a solution for this yet, but I think it deserves more thought and should be solved.  Otherwise it's very hard to form a proper mental model of the code.